### ISSUE TITLE
interaction of feature 'is_attributed' with the other categories

### ISSUE THREAD

**Issue by aranas** [mlp2018/fraud_detection#11]

**Comment by aranas** [comment]
as suggested by: https://www.kaggle.com/nanomathias/feature-engineering-importance-testing

"My thought is that some ips, apps, devices, etc. might have higher frequencies of is_attributed, and I wish to add that information, i.e. I'm calculating the following "attributed rates":

P(is_attributed|category)
 
or in some cases two- or multiple-paired combinations:

P(is_attributed|category_1, category_2)
 
The danger of this is that if a given category-combination has very few clicks, then the statistical significance of above equations cannot be trusted. Therefore I'll be weighing the rates by the following confidence rates:

confis_attributed=log(viewscategory_1)log(100000)
 
where the value 100000 has been chosen arbitrarily to such that if a given category has 1000 views, then it gets a confidence weight of 60%, if it has 100 views then onfly a confidence weight of 40% etc."

**Comment by aranas** [comment]
right now preprocessing is done before the cross-validation folding, however this means that new feature creation is in fact leaking data from validation set into training set (because features are created based on all data). Might also be problem for new features by grouping.
